Expand db path and improving visuals of multiline annotation text

cage2 opened the pull request:
Hi!

I have changed the loading and saving of annotations database to implements #89.

For each record of the database the file component  is abbreviated when saved and expanded (if required, i tried to not expand info nodes) when loaded.

Also implemented #90.  

Bye!
C.

Commit:
- stored abbreviated filenames  for the path component  of each record
  of the annotations database;
- improved a docstring.

Commit:
- extracted  local function  and taken  into account  info node  names
  (that should not be expanded).

Commit:
- added procedures to pad multiline annotation text.

Commit:
- wrapped 'annotate-wrap-annotation-in-box' with 'save-match-data.

Commit:
- fixed the width of the last row of the box

Commit:
- moved    'save-match-data'   from    the    function   that    calls
  'annotate--split-lines' inside  the body of the  latter; to remove
  any side-effect.

Commit:
- updated version;
- updated NEWS and Changelog (also fixed grammar for the former).

bastibe commented:
So many cool improvements! You pushed this package way farther than I'd ever hoped, @cage2!

cage2 commented:
On Mon, Dec 21, 2020 at 08:39:37AM -0800, Bastian Bechtold wrote:

Hi!

> So many cool improvements! You pushed this package way farther than I'd ever hoped, @cage2!

I just pushed a bunch of buttons.  :) Many useful helps came from some
kind   and  motivated   people  that   made  reviews,   testing,  gave
suggestions. ;-)

Bye!
C.

Commit:
- fixed integration of multiline annotated text.

Commit:
- removed code, in integration procedures, that should never runs.

Commit:
- fixed export for annotated text made from a single line.

Commit:
- rewritten export and integrate of annotations
  export  use  now the  Emacs  diff  functions  to generate  a  buffer
  containing the patch.

Commit:
- changed customizable variable related to exporting.

  As we are  using diff-mode could instruct the  executable to operate
  accordingly to a command line.

Commit:
- updated documentation.

Commit:
- fixed typos.

cage2 commented:
Hi! :)

If there are no problems i would merge this in the next hours.

Bye and happy new year! :)
C.

bastibe commented:
A happy new your to you as well! Merge away when you're ready (this is not a code review comment. I currently do not have the time to do code reviews, regrettably)!

cage2 commented:
On Sat, Jan 02, 2021 at 05:38:12AM -0800, Bastian Bechtold wrote:

Hi Bastian!

> A happy new your to you as well! Merge away when you're ready!

I'll merge in a minute!

In the meanwhile let me, again, truly wish you an happy new year!

Bye!
C.
